tools/rosgraph/test/test_roslogging.py

```diff
@@ -0,0 +1,55 @@
+# Software License Agreement (BSD License)
+#
+# Copyright (c) 2009, Willow Garage, Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer in the documentation and/or other materials provided
+#    with the distribution.
+#  * Neither the name of Willow Garage, Inc. nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import os
+
+from rosgraph import roslogging
+
+def test_configure_logging_makedirs_fails_wrong_grp():
+    """
+    Test proper error output if folder is wrong group.
+
+    Test that proper error output is given if `makedirs_with_parent_perms`
+    fails when ownership of folder cannot be changed.
+    """
+    test_folder = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            "test_logging_dir_root")
+    test_log = os.path.join(
+            test_folder,
+            "test_subfolder",
+            "test.log")
+    test_env = {"ROS_LOG_DIR": test_folder}
+    log_filename = roslogging.configure_logging("test", filename=test_log, env=test_env)
+    assert log_filename is None
+    assert os.path.isdir(test_folder)
+    os.rmdir(os.path.join(test_folder, "test_subfolder"))
```

Comment:
Tests shouldn't write into the source folder of the package. Please use a temporary path for that.